src/coord/src/coord/dataflow_builder.rs

```diff
@@ -106,7 +106,11 @@ impl<'a> DataflowBuilder<'a> {
             let index_oracle = self.index_oracle();
             let mut valid_indexes = index_oracle.indexes_on(*id).peekable();
             if valid_indexes.peek().is_some() {
-                for (index_id, idx) in valid_indexes {
+                // Deduplicate indexes by keys, in case we have redundant indexes.
+                let mut valid_indexes = valid_indexes.collect::<Vec<_>>();
+                valid_indexes.sort_by_key(|(_, idx)| &idx.keys);
+                valid_indexes.dedup_by_key(|(_, idx)| &idx.keys);
+                for (index_id, idx) in valid_indexes.into_iter() {
```

Comment:
```suggestion
                for (index_id, idx) in valid_indexes {
```

Reply:
Yeah, idk this is a style thing for me. Prevents me from having to figure out if `valid_indexes` is a `&[T]` or a `Vec<T>` to understand what the types are of things. I kinda feel most of Rust's cutesy elisions are bad. Can commit the change, but will grumble. :)

Reply:
Haha, it definitely was only a nit ;-)